userapi/storage/accounts/sqlite3/storage.go

```diff
@@ -281,6 +322,8 @@ func (d *Database) RemoveThreePIDAssociation(
 func (d *Database) GetLocalpartForThreePID(
 	ctx context.Context, threepid string, medium string,
 ) (localpart string, err error) {
+	d.threepidsMu.Lock()
+	defer d.threepidsMu.Unlock()
 	return d.threepids.selectLocalpartForThreePID(ctx, nil, threepid, medium)
```

Comment:
Read transactions do not need protection. They will never encounter or cause 'database is locked' errors. Only write transactions can. We don't need to mutex here and in other SELECT places.

Reply:
fixed